tezos/encoding-derive/src/make.rs

```diff
@@ -0,0 +1,689 @@
+            ))
+        }
+    };
+    Ok(StructEncoding { name, fields })
+}
+
+fn make_fields<'a>(
+    fields: impl IntoIterator<Item = &'a syn::Field>,
+) -> Result<Vec<FieldEncoding<'a>>> {
+    fields.into_iter().map(make_field).collect()
+}
+
+fn skip_field(meta: &[syn::Meta]) -> bool {
+    meta.iter().any(|meta| {
+        if let syn::Meta::Path(path) = meta {
+            path == symbol::SKIP
+        } else {
+            false
+        }
+    })
+}
+
+fn make_field(field: &syn::Field) -> Result<FieldEncoding> {
+    let meta = &mut get_encoding_meta(&field.attrs)?;
+    let skip = skip_field(meta);
+    let name = field.ident.as_ref().unwrap();
+    let encoding = if skip {
+        None
+    } else {
+        let encoding = make_type_encoding(&field.ty, meta)?;
+        let encoding = make_bounded_encoding(meta, encoding)?;
+        assert_empty_meta(meta)?;
+        Some(encoding)
+    };
+    Ok(FieldEncoding { name, encoding })
+}
+
+/// Creates encoding from the type `ty` and meta attributes.
+fn make_type_encoding<'a>(ty: &'a syn::Type, meta: &mut Vec<syn::Meta>) -> Result<Encoding<'a>> {
+    match ty {
+        syn::Type::Path(type_path) => make_type_path_encoding(&type_path.path, meta),
+        _ => Err(error_spanned(ty, "Unsupported type")),
+    }
+}
+
+/// Creates encoding from the type path `ty` (e.g. `mod::ty` or `u8`) and meta attributes.
+fn make_type_path_encoding<'a>(
+    path: &'a syn::Path,
+    meta: &mut Vec<syn::Meta>,
+) -> Result<Encoding<'a>> {
+    let segment = path.segments.last().unwrap();
+    match &segment.arguments {
+        syn::PathArguments::None => make_basic_encoding_from_type(&path, meta),
+        syn::PathArguments::AngleBracketed(args) => {
+            if args.args.len() != 1 {
+                return Err(error_spanned(&args.args, "Expected single argument"));
+            }
+            let arg = args.args.last().unwrap();
+            let inner_encoding = match arg {
+                syn::GenericArgument::Type(ty) => make_type_encoding(ty, meta)?,
+                _ => return Err(error_spanned(arg, "Only type generic parameters supported")),
+            };
+            let encoding = if segment.ident == symbol::rust::VEC {
+                make_list_encoding_from_type(path, meta, inner_encoding)?
+            } else if segment.ident == symbol::rust::OPTION {
+                make_optional_field_encoding_from_type(path, meta, inner_encoding)?
+            } else {
+                return Err(error_spanned(
+                    args,
+                    "Only `Vec<_>` and `Option<_>` are supported",
+                ));
+            };
+            let encoding = make_bounded_encoding(meta, encoding)?;
+            Ok(encoding)
+        }
+        _ => Err(error_spanned(
+            &segment.arguments,
+            "Only angle-bracketed generic arguments are supported",
+        )),
+    }
+}
+
+/// Constructs encoding for a non-parameterized type.
+///
+/// TODO: describe cases.
```

Comment:
@akoptelov 
what about this TODO:?
remove it or add TODO jire `TODO: TE-XYZ`  please

Reply:
Outdated comment. Removed.